builder/dockerfile/dispatchers.go

```diff
@@ -34,32 +31,14 @@ import (
 // Sets the environment variable foo to bar, also makes interpolation
 // in the dockerfile available from the next statement on via ${foo}.
 //
-func env(req dispatchRequest) error {
-	if len(req.args) == 0 {
-		return errAtLeastOneArgument("ENV")
-	}
-
-	if len(req.args)%2 != 0 {
-		// should never get here, but just in case
-		return errTooManyArguments("ENV")
-	}
-
-	if err := req.flags.Parse(); err != nil {
-		return err
-	}
-
-	runConfig := req.state.runConfig
+func (d *stageBuilder) dispatchEnv(c *instructions.EnvCommand) error {
```

Comment:
I come from a very object oriented background where we tend to group state and methods together in a coherent stuff and avoid free functions, so I am not sure I 100% agree, but for consistency sake, I'll do that.